File: App07_Movie/app/src/main/java/com/wooseok/movie/SecondFragment.java
package com.wooseok.movie;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.navigation.fragment.NavHostFragment;

import com.wooseok.movie.databinding.FragmentSecondBinding;

public class SecondFragment extends Fragment {

    private FragmentSecondBinding binding;

    @Override
    public View onCreateView(
            LayoutInflater inflater, ViewGroup container,
            Bundle savedInstanceState
    ) {

        binding = FragmentSecondBinding.inflate(inflater, container, false);

        // Activity에서 보내준 데이터를 fragment에서 받기
        if(getArguments() != null) {
            // 전달받은 데이터가 어떤 형태로 전송되어 오는지 확인하기
            Log.d("전달받은 데이터들",getArguments().toString());

            /*
                전달받은 변수중에 movie_search 변수가 있으면 데이터를
                getter 하여 movie_text 에 담아주고
                만약 데이터가 없으면 기본값으로 "없음" 이라는 문자열을
                movie_text에 담아 달라라
            */
            String movie_text = getArguments().getString("movie_search","없음");
            Log.d("검색문자열",movie_text);
        }

        return binding.getRoot();

    }

    public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }

}

File: HelloApp2/app/src/main/java/com/wooseok/hello/SecondFragment.kt
package com.wooseok.hello

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.navigation.fragment.findNavController
import com.wooseok.hello.databinding.FragmentSecondBinding

/**
 * A simple [Fragment] subclass as the second destination in the navigation.
 */
class SecondFragment : Fragment() {

    private var _binding: FragmentSecondBinding? = null

    // This property is only valid between onCreateView and
    // onDestroyView.
    private val binding get() = _binding!!

    override fun onCreateView(
            inflater: LayoutInflater, container: ViewGroup?,
            savedInstanceState: Bundle?
    ): View? {

        _binding = FragmentSecondBinding.inflate(inflater, container, false)
        return binding.root

    }
    /**
     * fragment 화면에서는
     * onCreateView() 에서는 root 를 binding 하고
     * 기타 view Component 의 event 핸들러 설정은
     * onViewCreate() 에서 설정한다.
     */
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        /**
         * 첫번째로 돌아가기 버튼을 클릭하면
         * nav_Graph.xml 에 설정된
         *      action_SecondFragment_to_firstFragment ID 찾아서
         *      destination 에 설정된 fragment 를 현재 화면에
         *      올려라
         */
        binding.buttonSecond.setOnClickListener {
            findNavController().navigate(R.id.action_SecondFragment_to_FirstFragment)
        }

        /**
         * 세번째 화면보기 버튼을 클릭하면
         * nav_Graph.xml 에 설정된
         *      action_SecondFragment_to_ThirdFragment 로 설정된
         *      ID 를 찾아서
         *      destination 으로 설정된 화면으로 교체하라
         */
        binding.buttonThird.setOnClickListener {
            findNavController().navigate(R.id.action_SecondFragment_to_ThirdFragment)
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}